src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -408,9 +408,13 @@ protected virtual async Task ExecuteNodeAsync(ExecutionContext context, Executio
                 if (result is not IDataLoaderResult)
                 {
                     CompleteNode(context, node);
-                    // for non-dataloader nodes that completed without throwing an error, we can re-use the context
-                    resolveContext.Reset(null, null);
-                    System.Threading.Interlocked.CompareExchange(ref context.ReusableReadonlyResolveFieldContext, resolveContext, null);
+                    // for non-dataloader nodes that completed without throwing an error, we can re-use the context,
+                    // except for enumerable lists - in case the user returned a value with LINQ that is based on the context source
```

Comment:
> that is based on the context source

maybe or maybe not

Reply:
This is merely a comment explaining the purpose of why this code was added.  Of course they may not have returned a LINQ enumerable at all.